packages/market-service/src/yearn/yearn-vaults.ts

```diff
@@ -250,17 +250,29 @@ export class YearnVaultMarketCapService implements MarketService {
       const vaultDayData: VaultDayData[] =
         response.data.account.vaultPositions[0].vault.vaultDayData
 
-      return vaultDayData.map((datum: VaultDayData) => {
-        return {
-          date: datum.timestamp,
-          price: bnOrZero(datum.tokenPriceUSDC)
-            .div(`1e+${USDC_PRECISION}`)
-            .times(datum.pricePerShare)
-            .div(`1e+${decimals}`)
-            .dp(6)
-            .toNumber()
+      return vaultDayData.reduce<HistoryData[]>((acc, current: VaultDayData) => {
+        const date = Number(current.timestamp)
+        const dateCheck = new Date(date).valueOf()
+        if (isNaN(dateCheck)) {
+          console.error('Yearn SDK vault has invalid date')
+          return acc
+        }
```

Comment:
I was trying to come up with a better naming for this like `isValidDate` but actually that's a bit weird since the validity check happens on next line. What about something like this?
```suggestion
        if (!isValidDate(new Date(date).valueOf())) {
          console.error('Yearn SDK vault has invalid date')
          return acc
        }
```
where `isValidDate` would be a type guard like so:

`const isValidDate = (timestamp: number) => !isNan(timestamp)`

Reply:
I like that. I wrote a util function for it.